v2/metadata/client.go

```diff
@@ -254,10 +254,11 @@ func (c *Client) CreateExecution(ctx context.Context, pipeline *Pipeline, taskNa
 	}
 
 	for _, id := range config.InputArtifactIDs {
+		thisId := id // declare a new variable, so we can take address of this
 		aePair := &pb.PutExecutionRequest_ArtifactAndEvent{
 			Event: &pb.Event{
 				Type:       pb.Event_INPUT.Enum(),
-				ArtifactId: &id,
+				ArtifactId: &thisId,
```

Comment:
this is a common golang gotcha, when we write `&id`, it refers to the loop variable, so it changes over time